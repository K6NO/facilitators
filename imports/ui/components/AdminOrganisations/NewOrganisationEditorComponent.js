/* eslint-disable max-len, no-return-assign */
import React from 'react';
import { Table, Button, ButtonGroup, Row, Col } from 'reactstrap';
import { Bert } from 'meteor/themeteorchef:bert';
// import './NewOrganisationEditorComponent.scss';

class NewOrganisationEditorComponent extends React.Component {
  constructor(props){
    super(props);
    this.state = {
        organisation : {
            name: '',
            billingAddress: '',
            billingCity: '',
            billingPostalCode: '',
            billingCountry: '',
            active: false,
          },
          orgActiveState: false,
    }
  }

  switchOrganisationState(value) {
    this.setState({
        ...this.state,
        orgActiveState : value,
    }); 
  }

  updateOrganisationState = (field, event) => {
    this.setState({
      organisation : {
        ...this.state.organisation,
        [field]: event.target.value,
      }
    });
  }

  saveNewOrganisation = () => {
      const { closeCallback } = this.props;
      const organisation = {
          name: this.state.organisation.name,
          billingAddress: this.state.organisation.billingAddress,
          billingCity: this.state.organisation.billingCity,
          billingPostalCode: this.state.organisation.billingPostalCode,
          billingCountry: this.state.organisation.billingCountry,
          active: this.state.organisation.active,
      }
      Meteor.call('organisations.insert', organisation, (error) => {
          if(error) {
              Bert.alert(error.reason, 'danger');
          } else {
              this.setState({
                organisation : {
                    name: '',
                    billingAddress: '',
                    billingCity: '',
                    billingPostalCode: '',
                    billingCountry: '',
                    active: false,
                  },
              });
              closeCallback();
          }
      });
  }

  render() {
    const { closeCallback } = this.props;
    return ( 
        <div className="NewOrganisationEditorComponent">
            <Row>
                <Col>
                <Table>
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Country</th>
                            <th>City</th>
                            <th>Address</th>
                            <th>Postal code</th>
                            <th></th>
                            <th></th>
                        </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>
                                    <input name="name" type="text" 
                                    onChange={(e) => this.updateOrganisationState('name', e)} 
                                    value={this.state.organisation.name} />
                                </td>
                                <td>
                                    <input name="country" type="text"                                         
                                    onChange={(e) => this.updateOrganisationState('billingCountry', e)} 
                                    value={this.state.organisation.billingCountry} />
                                </td>
                                <td>
                                    <input name="city" type="text" 
                                        onChange={(e) => this.updateOrganisationState('billingCity', e)} 
                                        value={this.state.organisation.billingCity} />
                                </td>
                                <td>
                                    <input name="city" type="text"                                         
                                        onChange={(e) => this.updateOrganisationState('billingAddress', e)} 
                                        value={this.state.organisation.billingAddress} />
                                </td>
                                <td>
                                    <input name="city" type="text" 
                                        onChange={(e) => this.updateOrganisationState('billingPostalCode', e)} 
                                        value={this.state.organisation.billingPostalCode} />
                                </td>
                                <td>
                                    <ButtonGroup>
                                        <Button color="primary" 
                                            onClick={() => this.switchOrganisationState(true)} 
                                            active={this.state.orgActiveState === true}>Active</Button>
                                        <Button color="primary" 
                                            onClick={() => this.switchOrganisationState(false)} 
                                            active={this.state.orgActiveState === false}>Paused</Button>
                                    </ButtonGroup>
                                </td>
                                <td>
                                    <ButtonGroup>
                                        <Button color="primary" onClick={() => this.saveNewOrganisation()}>Save</Button>
                                        <Button color="danger" onClick={closeCallback}>Cancel</Button>
                                    </ButtonGroup>
                                    
                                </td>
                            </tr>
                        </tbody>
                    </Table>
                </Col>
            </Row>
        </div>
    );
  }
}


export default NewOrganisationEditorComponent;