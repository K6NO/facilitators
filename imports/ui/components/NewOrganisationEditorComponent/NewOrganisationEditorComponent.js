/* eslint-disable max-len, no-return-assign */
import React from 'react';
import { Table, Button, ButtonGroup, Row, Col } from 'reactstrap';
import { Bert } from 'meteor/themeteorchef:bert';

class NewOrganisationEditorComponent extends React.Component {
  constructor(props){
    super(props);
    this.state = {
        organisation : {
            name: '',
            country: '',
            active: false,
          },
          orgActiveState: false,
    }
  }

  switchOrganisationState(value) {
    this.setState({
        ...this.state,
        orgActiveState : value,
    }); 
  }

  updateOrganisationState = (field, event) => {
    this.setState({
      organisation : {
        ...this.state.organisation,
        [field]: event.target.value,
      }
    });
  }

  saveNewOrganisation = () => {
      const { closeCallback } = this.props;
      const organisation = {
          name: this.state.organisation.name,
          country: this.state.organisation.country,
          active: this.state.organisation.active,
      }
      Meteor.call('organisations.insert', organisation, (error) => {
          if(error) {
              Bert.alert(error.reason, 'danger');
          } else {
              //reset the state
              this.setState({
                organisation : {
                    name: '',
                    country: '',
                    active: false,
                  },
              });
              closeCallback();
          }
      });
  }

  render() {
    const { closeCallback } = this.props;
    return ( 
        <div className="NewOrganisationEditorComponent">
            <Row>
                <Col>
                <Table>
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Country</th>
                            <th></th>
                            <th></th>
                        </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>
                                    <input name="name" type="text" 
                                    onChange={(e) => this.updateOrganisationState('name', e)} 
                                    value={this.state.organisation.name} />
                                </td>
                                <td>
                                    <input name="country" type="text"                                         
                                    onChange={(e) => this.updateOrganisationState('country', e)} 
                                    value={this.state.organisation.country} />
                                </td>
                                <td>
                                    <ButtonGroup>
                                        <Button color="primary" 
                                            onClick={() => this.switchOrganisationState(true)} 
                                            active={this.state.orgActiveState === true}>Active</Button>
                                        <Button color="primary" 
                                            onClick={() => this.switchOrganisationState(false)} 
                                            active={this.state.orgActiveState === false}>Inactive</Button>
                                    </ButtonGroup>
                                </td>
                                <td>
                                    <ButtonGroup>
                                        <Button color="primary" onClick={() => this.saveNewOrganisation()}>Save</Button>
                                        <Button color="danger" onClick={closeCallback}>Cancel</Button>
                                    </ButtonGroup>
                                    
                                </td>
                            </tr>
                        </tbody>
                    </Table>
                </Col>
            </Row>
        </div>
    );
  }
}


export default NewOrganisationEditorComponent;