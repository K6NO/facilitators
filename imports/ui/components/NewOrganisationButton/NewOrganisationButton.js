/* eslint-disable max-len, no-return-assign */
import React from 'react';
import { Table, Button, Row, Col } from 'reactstrap';
import NewOrganisationEditorComponent from '../AdminOrganisations/NewOrganisationEditorComponent';

// import './NewOrganisationButton.scss';

class NewOrganisationButton extends React.Component {
  constructor(props){
    super(props);
    this.state = {
        isOpen : false,
    }
  }

  openEditor = () => {
      if(!this.state.isOpen) {
          this.setState({
              isOpen: true,
          });
      }
  }

  closeEditor = () => {
      this.setState({
          isOpen: false,
      })
  }

  render() {
    return ( 
        <div className="NewOrganisationButton">
            <Row>
                <Col>
                    <Button 
                        color="primary"
                        onClick={this.openEditor}>Add Organisation
                    </Button>
                </Col>
            </Row>
            {this.state.isOpen ? (
                <Row>
                    <Col>
                        <NewOrganisationEditorComponent closeCallback={this.closeEditor} />
                    </Col>
                </Row>
            ) : ''}
        </div>
    );
  }
}


export default NewOrganisationButton;