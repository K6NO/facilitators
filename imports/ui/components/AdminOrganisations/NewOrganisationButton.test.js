import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Button } from 'reactstrap';
import NewOrganisationButton from './NewOrganisationButton';
import NewOrganisationEditorComponent from './NewOrganisationEditorComponent';


test('NewOrganisationButton renders', () => {
    const component = renderer.create(
        <NewOrganisationButton />
    ); 
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('NewOrganisationButton click calls function', () => {
    const mockFunction = jest.fn();
    const component = shallow(
        <Button 
            color="primary"
            onClick={mockFunction}>
            Add Organisation
        </Button>
    );
    const button = component.find('button').simulate('click');
    expect(button.length).toBe(1);
    expect(mockFunction.mock.calls.length).toEqual(1);
});

test('NewOrganisationButton state change displays NewOrganisationEditorComponent', () => {
    const component = shallow(
        <NewOrganisationButton />
    );
    component.setState({isOpen: true});
    expect(component.find(NewOrganisationEditorComponent).length).toBe(1);
});

test('NewOrganisationButton click changes state', () => {
    const component = shallow(
        <NewOrganisationButton />
    );
    component.setState({isOpen:false});
    const button = component.find(Button);
    button.simulate('click');
    expect(button.length).toBe(1);
    expect(component.state.isOpen).toBeTruthy;    
    expect(component.find(NewOrganisationEditorComponent).length).toBe(1);
});