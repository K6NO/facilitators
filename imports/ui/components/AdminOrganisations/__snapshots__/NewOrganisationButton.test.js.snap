// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewOrganisationButton renders 1`] = `
<div
  className="NewOrganisationButton"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Add Organisation
      </button>
    </div>
  </div>
  
</div>
`;
